arbos/programs/native.go

```diff
@@ -197,8 +268,10 @@ func getLocalAsm(statedb vm.StateDB, moduleHash common.Hash, addressForLogging c
 	zeroArbosVersion := uint64(0)
 	zeroGas := uint64(0)
 
+	targets := statedb.Database().WasmTargets()
 	// we know program is activated, so it must be in correct version and not use too much memory
-	info, asmMap, err := activateProgramInternal(statedb, addressForLogging, codeHash, wasm, pagelimit, program.version, zeroArbosVersion, debugMode, &zeroGas)
+	moduleActivationMandatory := false
```

Comment:
we might need to set `moduleActivationMandatory` here to true and activate the module each time we need to recompile to get the `activationInfo` (here `info`) to check `info.moduleHash` against `moduleHash`. 

alternatively, we can skip the check if info is nil, so we'd have:
```
if info != nil && info.moduleHash != moduleHash {
	log.Error("failed to reactivate program", "address", addressForLogging, "expected moduleHash", moduleHash, "got", info.moduleHash)
	return nil, fmt.Errorf("failed to reactivate program. address: %v, expected ModuleHash: %v", addressForLogging, moduleHash)
}
```

Reply:
there is also a question how it should work in context of the later code:
```
currentHoursSince := hoursSinceArbitrum(time)
if currentHoursSince > program.activatedAt {
	// stylus program is active on-chain, and was activated in the past
	// so we store it directly to database
	batch := statedb.Database().WasmStore().NewBatch()
	rawdb.WriteActivation(batch, moduleHash, asmMap)
	if err := batch.Write(); err != nil {
		log.Error("failed writing re-activation to state", "address", addressForLogging, "err", err)
	}
} else {
	// program activated recently, possibly in this eth_call
	// store it to statedb. It will be stored to database if statedb is commited
	statedb.ActivateWasm(info.moduleHash, asmMap)
}
```